### PR TITLE
Refactor extensions

### PR DESCRIPTION
**Pull request by pYr0x**
PR for #975

comments are welcome.

### CODE REVIEW

**Comment by matthewp** [comment]
Will review tomorrow, thanks!

**Comment by pYr0x** [comment]
we should think about system-trace. it is a system extension like all other. we should put system-trace into core.

**Comment by pYr0x** [comment]
Don't know why Travis failed, the error doesn't  say anything to me.

**Comment by matthewp** [comment]
@pYr0x agree on system-trace coming into core.

**Comment by pYr0x** [comment]
i can create a new issue and a PR for that

**Comment by matthewp** [comment]
Ok, as far as the CI breaking, looks like 1 test is failing in a few browsers: https://saucelabs.com/beta/builds/2bc0082fe93d454c9ba6b97c1d6fdf46

**Comment by pYr0x** [comment]
seems that the service worker test will fail

File: src/end.js [hunk 1]
```diff
@@ -21,7 +21,6 @@
 				}
 			});
 		global.steal.clone = cloneSteal;
-		global.steal.addSteal = addSteal;
```
Comment (matthewp) [inline src/end.js]: There's another addSteal in this file, is it not needed any more either?

Comment (pYr0x) [inline src/end.js]: i searched for `.addSteal` in the repo and didnt find another reference

**Comment by matthewp** [comment]
Would you mind pointing steal-tools to this branch and letting CI run on it, just to verify it doesn't mess with the build in any way?

**Comment by pYr0x** [comment]
@matthewp seems that steal-tools will work with this PR
https://github.com/stealjs/steal/tree/refactor-extensions
https://travis-ci.org/stealjs/steal-tools/builds/187590727

**Pull request merged**